src/pageEditor/hooks/useCreateModFromMod.ts

```diff
@@ -49,102 +51,87 @@ type UseCreateModFromModReturn = {
+        const newModId = newModMetadata.id;
 
-        try {
-          const newModDefinition = await buildAndValidateMod({
-            sourceMod: modDefinition,
-            cleanModComponents,
-            dirtyModComponentFormStates,
-            dirtyModOptionsDefinition,
-            dirtyModMetadata: metadata,
-          });
+        const unsavedModDefinition = await buildAndValidateMod({
+          sourceModDefinition,
+          draftModComponents,
+          // eslint-disable-next-line security/detect-object-injection -- new mod IDs are sanitized in the form validation
+          dirtyModOptionsDefinition: dirtyModOptionsDefinitionsMap[sourceModId],
+          dirtyModMetadata: newModMetadata,
+        });
 
-          const upsertResponse = await createMod({
-            modDefinition: newModDefinition,
-            organizations: [],
-            public: false,
-          }).unwrap();
-
-          const savedModDefinition: ModDefinition = {
-            ...newModDefinition,
-            sharing: {
-              public: upsertResponse.public,
-              organizations: upsertResponse.organizations,
-            },
-            updated_at: upsertResponse.updated_at,
-          };
-
-          if (!keepLocalCopy) {
-            await deactivateMod({ modId, shouldShowConfirmation: false });
-          }
-
-          const modComponents = [
-            ...dirtyModComponentFormStates,
-            ...cleanModComponents,
-          ];
-
-          dispatch(
-            modComponentActions.activateMod({
-              modDefinition: savedModDefinition,
-              configuredDependencies:
-                collectExistingConfiguredDependenciesForMod(
-                  savedModDefinition,
-                  modComponents,
-                ),
-              optionsArgs: collectModOptions(modComponents),
-              screen: "pageEditor",
-              isReactivate: false,
-            }),
-          );
-          dispatch(
-            editorActions.setActiveModId(savedModDefinition.metadata.id),
-          );
-          dispatch(editorActions.checkAvailableActivatedModComponents());
-
-          reportEvent(Events.PAGE_EDITOR_MOD_CREATE, {
-            copiedFrom: modId,
-            modId: savedModDefinition.metadata.id,
+        const upsertResponse = await createModDefinitionOnServer({
+          modDefinition: unsavedModDefinition,
+          ...createPrivateSharing(),
+        }).unwrap();
+
+        await updateReduxForSavedModDefinition({
+          // In the future, could consider passing the source mod id here if keepLocalCopy is false so that Page
+          // Editor navigation state is preserved for the source mod form states
+          modIdToReplace: undefined,
+          modDefinition: mapModDefinitionUpsertResponseToModDefinition(
+            unsavedModDefinition,
+            upsertResponse,
+          ),
+          draftModComponents,
+          isReactivate: false,
+        })(dispatch);
```

Comment:
There's an additional change we can make to be able to write this as: 

```suggestion
        await dispatch(
          updateReduxForSavedModDefinition({
            // In the future, could consider passing the source mod id here if keepLocalCopy is false so that Page
            // Editor navigation state is preserved for the source mod form states
            modIdToReplace: undefined,
            modDefinition: mapModDefinitionUpsertResponseToModDefinition(
              unsavedModDefinition,
              upsertResponse,
            ),
            draftModComponents,
            isReactivate: false,
          }),
        );
```

but I'll take care of that in a follow-on